core-primitives/stf-primitives/src/traits.rs

```diff
@@ -47,3 +49,26 @@ pub trait TrustedCallVerification {
 pub trait PoolTransactionValidation {
 	fn validate(&self) -> Result<ValidTransaction, TransactionValidityError>;
 }
+
+/// Trait to be implemented on the executor to serve helper methods of the executor
+/// to the `IndirectDispatch` implementation.
+pub trait IndirectExecutor<TCS, Error>
+where
+	TCS: PartialEq + Encode + Decode + Debug + Clone + Send + Sync + TrustedCallVerification,
+{
+	fn submit_trusted_call(&self, shard: ShardIdentifier, encrypted_trusted_call: Vec<u8>);
+
+	fn decrypt(&self, encrypted: &[u8]) -> Result<Vec<u8>, Error>;
+
+	fn encrypt(&self, value: &[u8]) -> Result<Vec<u8>, Error>;
+
+	fn get_enclave_account(&self) -> Result<AccountId, Error>;
+
+	fn get_default_shard(&self) -> ShardIdentifier;
+
+	fn sign_call_with_self<TC: Encode + Debug + TrustedCallSigning<TCS>>(
```

Comment:
It will probably also help if `TrustedCallSigning` is not generic over `TCS`, but has it as an associated type.

Reply:
I tried now, but the rabbit hole simply looks too deep: 
[ab/try-associated-types-for-trusted-call](https://github.com/integritee-network/worker/tree/ab/try-associated-types-for-trusted-call)

Reply:
Agree, this could be quite the hassle. Hard to foresee that, sorry for make you going through that